commit 98028cbd6903303f2551332e1dcfe3d801955910
Author: gwindlord <gwindlord@users.noreply.github.com>
Date:   Sat Oct 29 16:09:35 2016 +0200

    Revert "Disable unsupported sign-in and sync"
    
    This reverts commit 9e0363e4c8fc8efdaedc92a56410ffa3a425990f.
    
    Conflicts:
    	chrome/android/javatests/src/org/chromium/chrome/browser/signin/SigninTest.java

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkPromoHeader.java b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkPromoHeader.java
index 4dc5939..5cb69e7 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkPromoHeader.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/bookmarks/BookmarkPromoHeader.java
@@ -10,10 +10,8 @@ import android.support.v7.widget.RecyclerView;
 import android.support.v7.widget.RecyclerView.ViewHolder;
 import android.view.ViewGroup;
 
-import org.chromium.base.CommandLine;
 import org.chromium.base.ContextUtils;
 import org.chromium.base.metrics.RecordUserAction;
-import org.chromium.chrome.browser.ChromeSwitches;
 import org.chromium.chrome.browser.signin.SigninAccessPoint;
 import org.chromium.chrome.browser.signin.SigninAndSyncView;
 import org.chromium.chrome.browser.signin.SigninManager;
@@ -132,9 +130,7 @@ class BookmarkPromoHeader implements AndroidSyncSettingsObserver,
                 && mSignInManager.isSignInAllowed()
                 && !wasSigninPromoDeclined()
                 && ContextUtils.getAppSharedPreferences().getInt(
-                        PREF_SIGNIN_PROMO_SHOW_COUNT, 0) < MAX_SIGNIN_PROMO_SHOW_COUNT
-                && CommandLine.getInstance()
-                .hasSwitch(ChromeSwitches.ENABLE_SUPPRESSED_CHROMIUM_FEATURES);
+                        PREF_SIGNIN_PROMO_SHOW_COUNT, 0) < MAX_SIGNIN_PROMO_SHOW_COUNT;
         if (oldIsShowing != mShouldShow && notifyUI) {
             mShowingChangeListener.onPromoHeaderShowingChanged(mShouldShow);
         }
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/ntp/RecentTabsRowAdapter.java b/chrome/android/java/src/org/chromium/chrome/browser/ntp/RecentTabsRowAdapter.java
index 845f006..bfdcd8a 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/ntp/RecentTabsRowAdapter.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/ntp/RecentTabsRowAdapter.java
@@ -22,10 +22,8 @@ import android.widget.TextView;
 
 import org.chromium.base.ApiCompatibilityUtils;
 import org.chromium.base.metrics.RecordHistogram;
-import org.chromium.base.CommandLine;
 import org.chromium.base.metrics.RecordUserAction;
 import org.chromium.chrome.R;
-import org.chromium.chrome.browser.ChromeSwitches;
 import org.chromium.chrome.browser.favicon.FaviconHelper.FaviconImageCallback;
 import org.chromium.chrome.browser.ntp.ForeignSessionHelper.ForeignSession;
 import org.chromium.chrome.browser.ntp.ForeignSessionHelper.ForeignSessionTab;
@@ -888,19 +886,16 @@ public class RecentTabsRowAdapter extends BaseExpandableListAdapter {
             addGroup(new CurrentlyOpenTabsGroup(tabList));
         }
         addGroup(mRecentlyClosedTabsGroup);
-        if (CommandLine.getInstance()
-                .hasSwitch(ChromeSwitches.ENABLE_SUPPRESSED_CHROMIUM_FEATURES)) {
-            for (ForeignSession session : mRecentTabsManager.getForeignSessions()) {
-                if (!mHasForeignDataRecorded) {
-                    RecordHistogram.recordEnumeratedHistogram("HistoryPage.OtherDevicesMenu",
-                            OtherSessionsActions.HAS_FOREIGN_DATA, OtherSessionsActions.LIMIT);
-                    mHasForeignDataRecorded = true;
-                }
-                addGroup(new ForeignSessionGroup(session));
-            }
-            if (mRecentTabsManager.shouldDisplaySyncPromo()) {
-                addGroup(new SyncPromoGroup());
+        for (ForeignSession session : mRecentTabsManager.getForeignSessions()) {
+            if (!mHasForeignDataRecorded) {
+                RecordHistogram.recordEnumeratedHistogram("HistoryPage.OtherDevicesMenu",
+                        OtherSessionsActions.HAS_FOREIGN_DATA, OtherSessionsActions.LIMIT);
+                mHasForeignDataRecorded = true;
             }
+            addGroup(new ForeignSessionGroup(session));
+        }
+        if (mRecentTabsManager.shouldDisplaySyncPromo()) {
+            addGroup(new SyncPromoGroup());
         }
 
         // Add separator line after the recently closed tabs group.
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/preferences/MainPreferences.java b/chrome/android/java/src/org/chromium/chrome/browser/preferences/MainPreferences.java
index b3f8821..e21750f 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/preferences/MainPreferences.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/preferences/MainPreferences.java
@@ -17,7 +17,6 @@ import org.chromium.base.VisibleForTesting;
 import org.chromium.blimp_public.BlimpClientContext;
 import org.chromium.chrome.R;
 import org.chromium.chrome.browser.ChromeFeatureList;
-import org.chromium.chrome.browser.ChromeSwitches;
 import org.chromium.chrome.browser.PasswordUIView;
 import org.chromium.chrome.browser.autofill.PersonalDataManager;
 import org.chromium.chrome.browser.blimp.BlimpClientContextFactory;
@@ -189,13 +188,8 @@ public class MainPreferences extends PreferenceFragment implements SignInStateOb
 
     private void setupSignInPref() {
         mSignInPreference = (SignInPreference) findPreference(PREF_SIGN_IN);
-        if (!CommandLine.getInstance()
-                .hasSwitch(ChromeSwitches.ENABLE_SUPPRESSED_CHROMIUM_FEATURES)) {
-            getPreferenceScreen().removePreference(mSignInPreference);
-        } else {
-            mSignInPreference.registerForUpdates();
-            mSignInPreference.setEnabled(true);
-        }
+        mSignInPreference.registerForUpdates();
+        mSignInPreference.setEnabled(true);
     }
 
     private void clearSignInPref() {
diff --git a/chrome/android/javatests/src/org/chromium/chrome/browser/signin/SigninTest.java b/chrome/android/javatests/src/org/chromium/chrome/browser/signin/SigninTest.java
index f8c90d65..52fd33b 100644
--- a/chrome/android/javatests/src/org/chromium/chrome/browser/signin/SigninTest.java
+++ b/chrome/android/javatests/src/org/chromium/chrome/browser/signin/SigninTest.java
@@ -13,7 +13,6 @@ import android.content.DialogInterface;
 import android.preference.Preference;
 import android.support.v7.app.AlertDialog;
 import android.test.suitebuilder.annotation.MediumTest;
-import android.test.suitebuilder.annotation.Suppress;
 import android.widget.Button;
 
 import org.chromium.base.ThreadUtils;
@@ -287,7 +286,6 @@ public class SigninTest extends ChromeTabbedActivityTestBase {
     }
 
     @MediumTest
-    @Suppress
     @Restriction(ChromeRestriction.RESTRICTION_TYPE_GOOGLE_PLAY_SERVICES)
     public void testConsumerSignin() throws InterruptedException {
         SigninTestUtil.addTestAccount();
